SEI 2.0 Onboarding - Workflow + Set up + Analytics (#10887)
harness/developer-hub: docs/software-engineering-insights/harness-sei/get-started/sei-key-concepts.md

---
title: SEI 2.0 - Key concepts
description: The new experience for measuring engineering insights in Harness SEI
sidebar_label: Key concepts
sidebar_position: 10
---

> Harness Software Engineering Insights (SEI) 2.0 is a complete re-imagination of how engineering organizations can understand, measure, and improve software delivery. It brings together signals across your SDLC, connects them with organizational context, and transforms them into actionable insights for teams, managers, and executives.

Reviewer: ```suggestion
Harness Software Engineering Insights (SEI) 2.0 is a complete re-imagination of how engineering organizations can understand, measure, and improve software delivery. It brings together signals across your software delivery lifecycle (SDLC), connects them with organizational context, and transforms them into actionable insights for teams, managers, and executives.
```

:::note Coming Soon
Multi-level groupings based on custom attributes (e.g., Department → Function → Team) will be supported in future iterations. This will allow organizations to model more complex hierarchies and operational structures, making it easier to analyze engineering performance across departments, locations, and lines of business.
:::

### Teams

Teams are the core unit of measurement in SEI 2.0. Every leaf node in the Org Tree is treated as a Team, making it the fundamental grouping for surfacing insights, applying goals, and driving accountability.

Each Team represents a group of developers working together within the organizational hierarchy and is automatically derived from the Org Tree structure.

#### Key Characteristics

* **Auto-derived:** Every leaf node in the Org Tree is automatically considered a Team.
* **Configurable:** Each Team has its own configuration to define how metrics are calculated and displayed.
* **Contextualized & metric specific settings:** Teams include metadata such as associated integrations, relevant services, repositories, destination branches, pipelines, environments etc.

> This configuration layer adds precision to how metrics like Deployment Frequency, Lead Time, MTTR, and others are calculated ensuring each metric reflects the real scope, velocity, and complexity of the team's delivery lifecycle.

Reviewer: ```suggestion
This configuration layer adds precision to how metrics like Deployment Frequency, Lead Time, MTTR, and others are calculated, which ensures each metric reflects the real scope, velocity, and complexity of the team's delivery lifecycle.
```

### Profiles

> **Profiles** control **what SEI measures** and **how it measures it** by enabling or disabling specific metrics and defining which event types power those metrics. They provide a flexible way to customize SEI insights to your organization’s goals and workflows.

Reviewer: ```suggestion
**Profiles** control **what SEI 2.0 measures** and **how it measures it** by enabling or disabling specific metrics and defining which event types power those metrics. They provide a flexible way to customize SEI 2.0 insights to your organization’s goals and workflows.
```

Each Profile includes which **metrics** are enabled or disabled (e.g., Deployment Frequency, Lead Time, Change Failure Rate) & the **events and signals** that power those metrics.

#### Types of Profiles

SEI 2.0 supports three main types of Profiles, each optimized for different measurement objectives:

| Type of profile      | Purpose                                  | Metrics                                                  |
|---------------------|---------------------------------------------|-----------------------------------------------------------------|
| Efficiency      | Delivery performance (DORA metrics) & sprint insights (coming soon) | Deployment Frequency, Lead Time for Changes, Change Failure Rate, MTTR |
| Productivity    | Developer productivity and collaboration     | PR Velocity per Dev, PR Cycle Time,  Work Completed Per Dev, Coding Days Per Dev, Number of Comments Per PR, Average Time to First Comment  |
| Business Alignment | Alignment of engineering output to business goals | Business Alignment  |

#### Relationship to Org Tree & Teams

> - Teams define **where the data comes from** (repositories, pipelines, services etc).

Reviewer: ```suggestion
- Teams define **where the data comes from** (repositories, pipelines, services, etc).
```
